### DISCUSSION TITLE
Create toolbar app on MacOS

### DISCUSSION THREAD

**Discussion by FrankFrank9** [lukas-blecher/LaTeX-OCR#197] [Ideas]
Would improve user experience a lot making snipping much easier without calling everything from terminal!
Possible feature: 
-saving snip and predictions taken over time

**Comment by lukas-blecher** [comment]
True, but I have no idea how that could be achieved. There is a solution for linux in `pix2tex/setup_desktop.py` (but I think it's broken at the moment)
> saving snip and predictions taken over time

Yes, that is a good idea.

**Reply by FrankFrank9** [reply]
Give it a look

https://rumps.readthedocs.io/en/latest/

it would be interesting because one can pass functions from gui.py to create the toolbar without rewriting from scratch

**Reply by lukas-blecher** [reply]
Looks good. Looks like it can be an easy integration too. 
Much of the code is also already present for the api (https://github.com/lukas-blecher/LaTeX-OCR-PWC/blob/main/pix2tex/api/app.py)

**Reply by FrankFrank9** [reply]
Yup, I would use same user interface given in gui.py, just everything moved to a toolbar app

**Comment by FrankFrank9** [comment]
@lukas-blecher  any idea on how to call the gui
`latexocr`
rather than from the terminal?

**Reply by lukas-blecher** [reply]
I don't know what you mean

**Reply by FrankFrank9** [reply]
is there any other possible way to snip with the gui rather than going in terminal and writing `latexocr`?

**Reply by lukas-blecher** [reply]
An equivalent command would be `python -m pix2tex.__main__ --gui` but that's probably not what you were talking about.

In pure python there is a hacky way
```python
import sys
from pix2tex.__main__ import main

sys.argv[0]='latexocr'
main()
```
Hope this is what you're looking for

**Comment by FrankFrank9** [comment]
Please check out this first attempt of the toolbar app I've created
[https://github.com/FrankFrank9/LaTeX-Snipper](url)
the LaTeX-Snipper.py can be run itself.
Moreover, I would ask you to try compiling the standalone application as I'm getting errors due to arm64 (?). To do so, in the cloned folder run:
`pip install py2app`
`python setup.py py2app`

If this does not compile try alias mode as:
`pip install py2app`
`python setup.py py2app -A`

Just a first attempt, many improvements can be done

Install missing packages: presumably `rumps`

**Reply by lukas-blecher** [reply]
I won't be able to test it because I don't have a mac book.
